connection.go

```diff
@@ -251,12 +254,18 @@ func (ch *Channel) newConnection(conn net.Conn, initialState connectionState, ev
 	c.log = log
 	c.inbound.onRemoved = c.checkExchanges
 	c.outbound.onRemoved = c.checkExchanges
+	c.inbound.onAdded = c.onExchangeAdded
```

Comment:
should we remove the intermediate `onExchangeAdded` and instead directly make this call `callOnExchangeChange`?